source/manual/architecture-deep-dive.html.md

```diff
@@ -400,33 +400,41 @@ and processes.
 
 Puppet runs in a master/agent setup. There is a single "puppetmaster" running
 on its own class of machine, whereas the agents run on all the other machines
-(irrespective of class). Each puppet agent is responsible for sending alerts
-and logs to our various [monitoring systems][], and for running health check
-tests against the GOV.UK applications running on the node. The puppet master
-is in charge of keeping all of the puppet agents in sync with itself.
+(irrespective of class). Each puppet agent is responsible for running
+[health check][] tests against the GOV.UK applications running on the node,
+and for ensuring the presence of certain files and processes. The puppet
+master is in charge of keeping all of the puppet agents in sync with itself.
+
+[Icinga][] alerts us to problems with our machines and apps. There is a wide
+variety of different alerts, all of which are configured in Puppet. For
+monitoring, 'filebeat' is used to send logs to [Logit][], and 'statsd' exports
+most monitoring metrics, which can be viewed in [Grafana][]. Read more about
+[tooling for monitoring][tools].
 
 If an app release contains a major change such as a renamed environment
 variable, then it will require an application restart, which would bring the
-application offline on that machine. The machine would fail its [healthcheck]
-and our load balancer would begin serving traffic from different nodes of the
-same class: for this reason it is important that all machines are updated at
-different times. Therefore, each puppet instance
-[runs itself every half hour][puppet-cronjob], with the puppet agents
-configured to run after the puppetmaster _at randomised times_. 
-
-What actually happens when puppet runs itself? The puppetmaster copies the
-latest versions of govuk-puppet and govuk-secrets to itself (this is why
-you must wait 30 minutes between [deploying Puppet][] between environments).
-The puppet agents, on the other hand, compare their configuration with the
+application offline on that machine. The load balancer would begin serving
+traffic from different nodes of the same class: for this reason it is
+important that all machines are updated at different times. Therefore, each
+instance [runs puppet every half hour][puppet-cronjob], with the puppet
+agents configured to run after the puppetmaster at randomised times.
+
+What actually happens on a puppet run? The puppetmaster copies the latest
```

Comment:
The copying of govuk-puppet and govuk-secrets to the Puppetmaster happens when you run the Deploy Puppet job on Jenkins - I wouldn't say that's part of the "puppet run" (puppet agent run) as such.

Reply:
Thanks, I'll amend this. So what does the 'puppet agent run' on the puppetmaster machine do? Does it just detect that it's the master and skip itself?

Reply:
The Puppet agent works in just the same way on the master as it does on all the other machines. That is, it checks for differences between what it sees and what the Puppetmaster (which just happens to be itself) says should be there. Just like you've described at the end of the paragraph :) Or in other words, Puppet manages the Puppetmaster just like any other machine.

Reply:
Thanks! Have amended.